packages/cashier/src/Stores/__tests__/on-ramp-store.spec.js

```diff
@@ -0,0 +1,377 @@
+    });
+
+    it('should set should_show_widget into false if html widget is not defined when disposeGetWidgetHtmlReaction reaction is running', async () => {
+        const spySetShouldShowWidget = jest.spyOn(on_ramp_store, 'setShouldShowWidget');
+        on_ramp_store.setSelectedProvider(changelly_provider);
+        changelly_provider.getWidgetHtml = jest.fn().mockResolvedValueOnce('');
+        on_ramp_store.onMountOnramp();
+        on_ramp_store.setShouldShowWidget(true);
+
+        await waitFor(() => {
+            expect(spySetShouldShowWidget).toHaveBeenCalledWith(false);
+        });
+    });
+
+    it('should set widget error if there is an error when requesting widget when disposeGetWidgetHtmlReaction reaction is running', async () => {
+        const spySetWidgetError = jest.spyOn(on_ramp_store, 'setWidgetError');
+        on_ramp_store.setSelectedProvider(changelly_provider);
+        changelly_provider.getWidgetHtml = jest.fn().mockRejectedValueOnce('Request error');
+        on_ramp_store.onMountOnramp();
+        on_ramp_store.setShouldShowWidget(true);
+
+        await waitFor(() => {
+            expect(spySetWidgetError).toHaveBeenCalledWith('Request error');
+        });
+    });
+
+    it('should not call setIsRequestingWidgetHtml method if is_requesting_widget_html already equal to true when disposeGetWidgetHtmlReaction reaction is running', () => {
+        const spySetIsRequestingWidgetHtml = jest.spyOn(on_ramp_store, 'setIsRequestingWidgetHtml');
+        on_ramp_store.is_requesting_widget_html = true;
+        on_ramp_store.setSelectedProvider(changelly_provider);
+        on_ramp_store.onMountOnramp();
+        on_ramp_store.setShouldShowWidget(true);
+
+        expect(spySetIsRequestingWidgetHtml).not.toHaveBeenCalled();
+    });
+
+    it('should call disposeThirdPartyJsReaction and disposeGetWidgetHtmlReaction reactions when unmount onramp', () => {
+        on_ramp_store.onMountOnramp();
+        const spyDisposeThirdPartyJsReaction = jest.spyOn(on_ramp_store, 'disposeThirdPartyJsReaction');
+        const spyDisposeGetWidgetHtmlReaction = jest.spyOn(on_ramp_store, 'disposeGetWidgetHtmlReaction');
+        on_ramp_store.onUnmountOnramp();
+
+        expect(spyDisposeThirdPartyJsReaction).toBeCalledTimes(1);
+        expect(spyDisposeGetWidgetHtmlReaction).toBeCalledTimes(1);
+    });
+
+    it('should show and hide deposit address popover when deposit address is copied', async () => {
+        jest.useFakeTimers();
+        jest.spyOn(document, 'createRange').mockImplementation(() => ({
+            selectNodeContents: jest.fn(),
+        }));
+        jest.spyOn(window, 'window', 'get').mockImplementation(() => ({
+            getSelection: () => ({
+                addRange: jest.fn(),
+                removeAllRanges: jest.fn(),
+            }),
+        }));
+        Object.assign(navigator, {
+            clipboard: {
+                writeText: jest.fn().mockImplementation(() => Promise.resolve()),
```

Comment:
We can use the shorthand for this:
```suggestion
                writeText: jest.fn(() => Promise.resolve()),
```

Reply:
@farrah-deriv sure, thx for the notice